pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -215,40 +215,50 @@ func (os *OpenStack) AttachDisk(instanceID, volumeID string) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
 	if volume.Status != VolumeAvailableStatus {
```

Comment:
Sorry, and like I said; perhaps I'm confused here.  My point was that I'm unclear on the checks that follow this.  We're performing a fetch of the volume here from cinder directly, if that volume is NOT available we err out and return.  So what I'm not sure I follow is the check on L#225 and some of the logic that follows it, and then what seems to be a redundant call to DiskIsAttached?  What I'm saying is that if Cinder/Gophercloud is returning a volume with a status of "available" then there should be no reason to inspect the attachments entries for that volume, it's not attached, and the DiskIsAttached check seems confusing to me here because... well; we basically just performed that check.  Am I missing something obvious here?